lib/msprime.c

```diff
@@ -2116,6 +2119,35 @@ msp_common_ancestor_event(msp_t *self, population_id_t population_id)
 }
 
 
+/* This calculates the rate given by Eq (2) in the notes
+ */
+static double
+msp_compute_lambda_Xi_dirac(unsigned int b, double psi, double c)
+{
+    assert(b > 0);
+    assert(psi > 0);
+    assert(psi < 1);
+    assert(c >= 0);
+
+    unsigned int l, n_block;
+    n_block = GSL_MIN(b, 4);
+    double ret = 0;
```

Comment:
Keeping code and declarations separate is a good idea for portability (and clarity IMO). Here I would have
```c
unsigned int l, n_block;
double ret = 0;

assert(b > 0);
￼assert(psi > 0);
￼assert(psi < 1);
￼assert(c >= 0);

n_block = GSL_MIN(b, 4);